### DISCUSSION TITLE
Run in Ubuntu

### DISCUSSION THREAD

**Discussion by fabioselau077** [omkarcloud/google-maps-scraper#33] [Q&A]
Using my macOS I just ran 3 commands and everything was working perfectly, but when uploading to an EC2 Ubuntu the problems started to appear.
Here is a mini tutorial for those who have the same problems, I tested it with Python 3.9 and 3.10.
1. `apt-get install python3-pip` (if not install)

2. `git clone https://github.com/omkarcloud/google-maps-scraper`

3. `cd google-maps-scraper`

4. `python3 -m pip install -r requirements.txt`

5. Download and move chromedriver (file not folder...) (https://github.com/omkarcloud/google-maps-scraper/issues/20#issuecomment-1659920534) to /build if not exists

6. “`chmod 755`” in build/chromedriver

7. `apt update && apt install wget` (if not install)

8. `wget https://dl.google.com/linux/chrome/deb/pool/main/g/google-chrome-stable/google-chrome-stable_114.0.5735.198-1_amd64.deb`

9. `apt install ./google-chrome-stable_114.0.5735.198-1_amd64.deb`

10. `python3 main.py`

**Comment by Chetan11-dev** [comment]
If you are encountering this issue, please note that it is advisable to run the program in Docker rather than on VPS/Cloud instances. You can find the instructions for running it in Docker [here](https://github.com/omkarcloud/google-maps-scraper#-how-to-run-it-in-docker).

**Reply by danielpbrewer** [reply]
what or where is the 'build' directory?  are you ssaying to create it in the root directory of the google scraper if it doesn't exist?

**Reply by Chetan11-dev** [reply]
We only support running in Google Cloud VM, Kindly follow steps at https://github.com/omkarcloud/botasaurus#how-to-run-scraper-in-virtual-machine, if you want to do so.
